### PR TITLE
Adapt IdentityValidator with test cases

### PR DESCRIPTION
**Pull request by Momo-Not-Emo**
The Identity Validator, based on Soot code, now includes added test cases covering all branches, except for cases where "parameterRef" is out-of-index. This omission is intentional, as it prevents the jimple file from being unloaded due to such jimple methods.

### CODE REVIEW

File: sootup.core/src/main/java/sootup/core/validation/IdentityValidator.java [hunk 1]
```diff
@@ -33,40 +40,65 @@
...
+     * @return
      */
-    return null;
-  }
+    @Override
+    public List<ValidationException> validate(Body body, View view) {
+        List<ValidationException> exceptions = new ArrayList<>();
+
+        boolean hasThisLocal = false;
+        SootMethod method = view.getMethod(body.getMethodSignature()).get();
```
Comment (JonasKlauke) [inline sootup.core/src/main/java/sootup/core/validation/IdentityValidator.java]: handle the case in which the method can not be found in the given view.

File: sootup.core/src/main/java/sootup/core/validation/IdentityValidator.java [hunk 2]
```diff
@@ -33,40 +40,65 @@
...
+                    if (ref.getIndex() < 0 || ref.getIndex() >= paramCount) {
+                        if (paramCount == 0) {
+                            exceptions.add(new ValidationException(id,
+                                    "This methodRef has no parameters, so no parameter reference is allowed"));
+                        } else {
+                            exceptions.add(new ValidationException(id,
+                                    String.format("Parameter reference index must be between 0 and %d (inclusive)",
+                                            paramCount - 1)));
+                        }
+                        return exceptions;
```
Comment (JonasKlauke) [inline sootup.core/src/main/java/sootup/core/validation/IdentityValidator.java]: if the method also has no this reference, it won't list the exceptions in the results

**Review by JonasKlauke — CHANGES_REQUESTED** [review]
Thanks for the help 👍 
I spotted two improvements. It would be nice if you could include them

File: sootup.core/src/main/java/sootup/core/validation/IdentityValidator.java (hunk 1 above)
Comment (Momo-Not-Emo) [reply sootup.core/src/main/java/sootup/core/validation/IdentityValidator.java]: > handle the case in which the method can not be found in the given view.

what's the expected behaviour when the method does not present in the given view? Skip this validator or throw an exception?

**Comment by Momo-Not-Emo** [comment]
> Thanks for the help 👍 I spotted two improvements. It would be nice if you could include them

@JonasKlauke I have made changes according to your review. Could you review the changes? Thx~

File: sootup.core/src/main/java/sootup/core/validation/IdentityValidator.java (hunk 1 above)
Comment (swissiety) [reply sootup.core/src/main/java/sootup/core/validation/IdentityValidator.java]: We should not fail - just abort - we can't say anything about it and in this case we should not complain as we could be in in a partial analysis situation - i.e. SootUp should do the most of the available information

**Review by JonasKlauke — APPROVED** [review]
Thanks 👍

**Comment by JonasKlauke** [comment]
Always run the fmt:format plugin before commiting or pushing 😄 I have also created so many fmt commits 😆

**Pull request merged**